### ISSUE TITLE
Enhancement: x64-bit compatibility 

### ISSUE THREAD

**Issue by SAS-1** [m1lkman/Update-PlexMediaServer#24]
Hi @m1lkman 

Not sure if you are still looking at this repo but if so can you amend it to download the X64-bit version if it's installed instead of the 32-bit version. 

I took a stab at it myself but have struggled to get it working. 

Thanks

**Comment by m1lkman** [comment]
@SAS-1, I hadn't looked at this script in so long, so glad you're still getting some use out of it. I moved away from hosting plex on Windows. 

Check out the development branch and let me know if that works for you. I added a few other things/fixes. If you run it on a 64-bit os it will now default to the x86_64 build. You can force it otherwise with the -Build parameter, eg `-Build 'windows-x86'`.

**Comment by SAS-1** [comment]
@m1lkman you are amazing, I have been looking at this off and on for a few months and never getting it quite right and you fix it in a few hours!

I ran a manual update with -force and it downloaded and I went through the install manually to check it and all looks great.
Really appreciate it!

**Comment by m1lkman** [comment]
@SAS-1 , great. Glad it's working for you. I'll probably do a few more small tweaks and push it to master over the next few days.

**Comment by m1lkman** [comment]
@SAS-1 , pushed a release with these changes and more. I did swap how the x64 support works in the interest of continuity. Unless you specify with the -Build parameter, the script will check for updates and install those that match the currently installed build. Please let me know if you have any issues.

**Comment by SAS-1** [comment]
Hi @m1lkman  
thanks for this, when I run it with force command I got a message about the token

```
Import-Module Update-PlexMediaServer
Update-PlexMediaServer -Force
Verifying Server Online Authentication Token...Server Online Token Authentication Failed
Please verify specified Plex Server Online Autentication Token or use Get-PlexToken to retrieve one.
```

reran the get token command and then ran the script - same error I then added the token to the command and got this

```
Update-PlexMediaServer : Cannot bind positional parameters because no names were given.
At line:3 char:1
+ Update-PlexMediaServer -UseServerToken TOKEN
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Update-PlexMediaServer], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousPositionalParameterNoName,Update-PlexMediaServer
```

Edit forgot to say rolling the 2 files back to yesterdays version works fine. 

Edit 2: Did some more testing something in the newest code is stopping it working, even removing the module, deleting the folder and reinstalling never fixed it. Yesterday's code is working I also noticed that it's not getting the hostname for some reason

```
WARNING: Exception determining active sessions. System.Collections.Hashtable[0].exception.message
Cannot bind parameter 'Uri'. Cannot convert value "http://:32400/livetv/sessions/?X-Plex-Token=TOKEN" to type "System.Uri". Error: "Invalid URI: The hostname could not be parsed."
WARNING: Exception determining Live TV/DVR sessions. System.Collections.Hashtable.exception.message
```
I even tried coding it as $HostName=IPAddress in line 161 but same error not had time to work out what is causing it yet though.

**Comment by m1lkman** [comment]
@SAS-1 ,

I'm not able to recreate any of these issues.

The issue at the top I believe is related to the script having issues finding the `PlexOnlineToken` in the registry. Is this machine you're testing on have plex signed in and claimed? It could also be finding an old token that's no longer value..maybe. Try using the `-verbose` switch and see if you can learn any more. If you can send me the verbose output and/or log (sanitize any tokens) I may be able to figure it out.

The second error is because the `-UseServerToken` parameter is just a switch parameter so it doesn't take a value. If you want to use a token instead of the server token try using the `-PlexToken 'xxxxxxxxxx'` parameter.

The third error I don't have a great theory on.  If you don't use the `-PlexServerHostname` parameter to specify a manual hostname it's going to use the same code it did before to do a reverse IP lookup and figure out your public hostname. Verbose output would help here also. Is the machine you're testing on just a test machine or is this your primary plex server? As a simple test, you could try `-PlexServerHostname localhost`. This would force the hostname value as I do for testing on my test system thats not publically available.

**Comment by SAS-1** [comment]
Hi

No worries, I will keep looking at it, it's likely something at my end then as soon as I swap the code to the new one I get the token error but swapping the code back it works fine. I will try the verbose and see if it shows anything. I can get it working with yesterday's code so I am OK with that.

Thanks anyway

**Comment by m1lkman** [comment]
The more I uninstall and reinstall the more I realize that the installer isn't great at removing old info and tokens from the registry. Any verbose output you can provide to help me make this script smarter will help. Thanks.

**Comment by m1lkman** [comment]
@SAS-1 , take a look at the latest dev branch. I think I figured it out. When we move between x64 & x86 builds some of the settings don't get removed and are also stored differently for each build. I improved the logic so it should pull the server token correctly now.

**Comment by SAS-1** [comment]
hi just tried the latest Dev branch and it worked for me thanks so much!

Verbose log if it helps you (it was sucessful and found the hostname this time though)

> 2023-03-11 20:00:56 INFO: Update-PlexMediaServer Starting
2023-03-11 20:00:56 INFO: Server Online Token Authentication execution enabled
2023-03-11 20:00:56 INFO: Force Update enabled via command-line (-Force)
2023-03-11 20:00:56 INFO: Plex Media Server Settings found in Registry [Count: 3]
2023-03-11 20:00:56 INFO: Plex Media Server installation found in C:\Program Files\Plex\Plex Media Server
2023-03-11 20:00:56 INFO: InstallFolder: C:\Program Files\Plex\Plex Media Server
2023-03-11 20:00:56 INFO: Plex Settings found in key HKEY_USERS\.DEFAULT\SOFTWARE\Plex, Inc.\Plex Media Server are missing values, Plex Media Server may not be logged in and claimed
2023-03-11 20:00:56 INFO: Plex Media Server installation found in C:\Program Files\Plex\Plex Media Server
2023-03-11 20:00:56 INFO: InstallFolder: C:\Program Files\Plex\Plex Media Server
2023-03-11 20:00:56 INFO: Plex Settings Key Found HKEY_USERS\GUID\SOFTWARE\Plex, Inc.\Plex Media Server
2023-03-11 20:00:56 INFO: UpdateChannel: Beta
2023-03-11 20:00:56 INFO: LocalAppDataPath: C:\Users\USER\AppData\Local\
2023-03-11 20:00:56 INFO: PlexOnlineMail: EMAIL
2023-03-11 20:00:56 INFO: PlexOnlineUsername: USERNAME
2023-03-11 20:00:56 INFO: PlexOnlineToken: Token
2023-03-11 20:00:57 INFO: Plex Server Online Authentication Token TOKEN Validated
2023-03-11 20:00:57 INFO: Plex Media Server executable C:\Program Files\Plex\Plex Media Server\Plex Media Server.exe is version 1.31.2.6783 configured to run as user 
2023-03-11 20:00:57 INFO: Plex Media Server process running C:\Program Files\Plex\Plex Media Server\Plex Media Server.exe in user context PCNAME\USER
2023-03-11 20:00:57 INFO: PCNAME\USER SID: GUID
2023-03-11 20:00:57 INFO: HostName is HostName
2023-03-11 20:00:58 INFO: Plex Media Server Service Wrapper (PlexService) found installed (Version: 1.2.1.0).
2023-03-11 20:00:58 INFO: Plex Media Server Build windows-x86_64 Detected
2023-03-11 20:00:58 INFO: Update version 1.31.2.6783-9209b39b4 available for download.
2023-03-11 20:00:58 INFO: Version up-to-date. Installed version (1.31.2.6783) equal to available version (1.31.2.6783).
2023-03-11 20:00:58 INFO: Proceeding with update. Force update enabled.
2023-03-11 20:00:59 INFO: Latest update file found with matching checksum (C:\Users\USER\AppData\Local\\Plex Media Server\Updates\1.31.2.6783-9209b39b4\Plex-Media-Server-1.31.2.6783-9209b39b4.exe)
2023-03-11 20:00:59 INFO: No active sessions found.
2023-03-11 20:00:59 INFO: No active Live TV/DVR Sessions found
2023-03-11 20:00:59 INFO: Found Plex Media Server Service Wrapper (PlexService) Running.
2023-03-11 20:01:00 INFO: Sent Plex Media Server Service Wrapper (PlexService) Stop-Service.
2023-03-11 20:01:01 INFO: Plex Media Server Service Wrapper (PlexService) Stopped.
2023-03-11 20:01:01 INFO: Plex Media Server processes found running.
2023-03-11 20:01:01 INFO: Sent Plex Media Server processes Stop-Process. (Plex Update Service PlexScriptHost PlexScriptHost)
2023-03-11 20:01:01 INFO: Sent Plex Media Server processes Stop-Process. (PlexScriptHost PlexScriptHost)
2023-03-11 20:01:01 INFO: Plex Media Server processes stopped.
2023-03-11 20:01:01 INFO: Starting Plex Media Server update Process: C:\Users\USER\AppData\Local\\Plex Media Server\Updates\1.31.2.6783-9209b39b4\Plex-Media-Server-1.31.2.6783-9209b39b4.exe /NORESTART
2023-03-11 20:01:40 INFO: Plex Media Server Settings found in Registry [Count: 3]
2023-03-11 20:01:40 INFO: Plex Media Server installation found in C:\Program Files\Plex\Plex Media Server
2023-03-11 20:01:40 INFO: InstallFolder: C:\Program Files\Plex\Plex Media Server
2023-03-11 20:01:40 INFO: Update successfully installed with ExitCode 0.
2023-03-11 20:01:41 INFO: Sent Plex Media Server Service Wrapper (PlexService) Start-Service.
2023-03-11 20:01:41 INFO: Plex Media Server Service Wrapper (PlexService) Started.
2023-03-11 20:01:41 INFO: Plex Media Server process running C:\Program Files\Plex\Plex Media Server\Plex Media Server.exe as User PCNAME\USER
2023-03-11 20:01:48 INFO: Update-PlexMediaServer Completed

**Comment by m1lkman** [comment]
Thanks @SAS-1  for your assistance, glad we got it figured out.
